cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsSearchHandler.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.metrics.query;
+
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.metrics.MetricsScope;
+import co.cask.cdap.data2.OperationException;
+import co.cask.cdap.gateway.auth.Authenticator;
+import co.cask.cdap.metrics.data.MetricsScanQuery;
+import co.cask.cdap.metrics.data.MetricsScanQueryBuilder;
+import co.cask.cdap.metrics.data.MetricsTableFactory;
+import co.cask.cdap.metrics.data.TimeSeriesTable;
+import co.cask.http.HttpResponder;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.google.inject.Inject;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedSet;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+/**
+ * Handler to list metrics top-level-contexts , get next-level contexts and obtain metrics at a given context.
+ */
+
+//todo : when MetricsQueryHandler class supports namespaces, these endpoints will conflict,
+// we might want to combine this logic into MetricsQueryHandler or change the endpoint here based on API Review.
```

Comment:
@abaranau , didn't add v2 handlers due to this conflict, based on review, it might be easier to change the endpoint prefix here or move it to query handler.